### PR TITLE
Fix documentation errors

### PR DESCRIPTION
**Pull request by beam2d**
This PR fixes errors on `make html` and `make doctest` of sphinx. It leaves some warnings caused by duplicated link target names (`[1]` appearing in multiple docstrings rendered in one page); they should be fixed by splitting classes into separate pages instead of fixing the docstrings themselves.

### CODE REVIEW

**Comment by delta2323** [comment]
Could you apply the change of #2759 as well ?

**Comment by beam2d** [comment]
Applied.

File: chainer/cuda.py [hunk 1]
```diff
@@ -181,8 +181,8 @@ def get_device(*args):
     .. note::
 
         This API is deprecated. Please use
-        :method:`cupy.cuda.get_device_from_id`
-        or :method:`cupy.cuda.get_device_from_array` instead.
+        :func:`~chainer.cuda.get_device_from_id`
```
Comment (delta2323) [inline chainer/cuda.py]: Ah, sorry. `:func:` should be `:meth:`. I mistakenly used the former one.

Comment (beam2d) [inline chainer/cuda.py]: I think `:func:` is correct here.

**Comment by beam2d** [comment]
I fixed the errors occurred on Sphinx 1.6.1.

File: chainer/cuda.py (hunk 1 above)
Comment (delta2323) [inline chainer/cuda.py]: Then, it is OK to leave it.

File: docs/source/tutorial/type_check.rst [hunk 2]
```diff
@@ -67,7 +67,7 @@ Otherwise this code throws an exception, and the user gets a message like this:
 
    Traceback (most recent call last):
    ...
-   InvalidType: Expect: in_types[0].ndim == 2
+   chainer.utils.type_check.InvalidType: Expect: in_types[0].ndim == 2
```
Comment (delta2323) [inline docs/source/tutorial/type_check.rst]: This change caused error in the doctest.

File: docs/source/tutorial/type_check.rst [hunk 3]
```diff
@@ -102,8 +102,8 @@ And an error is like this:
 
    Traceback (most recent call last):
    ...
-   InvalidType: Expect: in_types[0].dtype == <type 'numpy.float64'>
-   Actual: float32 != <type 'numpy.float64'>
+   chainer.utils.type_check.InvalidType: Expect: in_types[0].dtype == <class 'numpy.float64'>
```
Comment (delta2323) [inline docs/source/tutorial/type_check.rst]: ditto

File: docs/source/tutorial/type_check.rst [hunk 4]
```diff
@@ -162,7 +162,7 @@ When ``x_type.shape[0] == 3`` and ``y_type.shape[0] == 1``, users can get the er
 
    Traceback (most recent call last):
    ...
-   InvalidType: Expect: in_types[0].shape[0] == in_types[1].shape[0] * 4
+   chainer.utils.type_check.InvalidType: Expect: in_types[0].shape[0] == in_types[1].shape[0] * 4
```
Comment (delta2323) [inline docs/source/tutorial/type_check.rst]: ditto

File: docs/source/tutorial/type_check.rst [hunk 5]
```diff
@@ -222,7 +222,7 @@ The above example produces an error message like this:
 
    Traceback (most recent call last):
    ...
-   InvalidType: Expect: sum(in_types[0].shape) == 10
+   chainer.utils.type_check.InvalidType: Expect: sum(in_types[0].shape) == 10
```
Comment (delta2323) [inline docs/source/tutorial/type_check.rst]: ditto

File: docs/source/tutorial/type_check.rst [hunk 6]
```diff
@@ -255,7 +255,7 @@ This code generates the following error message:
 
    Traceback (most recent call last):
    ...
-   InvalidType: Expect: Shape is expected to be ...
+   chainer.utils.type_check.InvalidType: Expect: Shape is expected to be ...
```
Comment (delta2323) [inline docs/source/tutorial/type_check.rst]: ditto

File: docs/source/tutorial/type_check.rst [hunk 7]
```diff
@@ -102,8 +102,8 @@ And an error is like this:
 
    Traceback (most recent call last):
    ...
-   InvalidType: Expect: in_types[0].dtype == <type 'numpy.float64'>
-   Actual: float32 != <type 'numpy.float64'>
+   InvalidType: Expect: in_types[0].dtype == <class 'numpy.float64'>
+   Actual: float32 != <class 'numpy.float64'>
```
Comment (delta2323) [inline docs/source/tutorial/type_check.rst]: Could you revert this change as well?
